File: examples/basics/type_delay_test.js
const { remote } = require('webdriverio');

async function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

async function main() {
    const opts = {
        hostname: '172.16.1.52',
        port: 4723,
        path: '/',
        capabilities: {
            "appium:automationName": "NovaWindows2",
            "platformName": "Windows",
            "appium:app": "Root",
            "appium:typeDelay": 0 // Start with 0 delay
        },
        logLevel: 'error'
    };
    const driver = await remote(opts);
    try {
        const element = await driver.$("//Document[@Name='Text Editor']");

        console.log("1. Typing with default configured delay (0ms) - Should be fast");
        await element.setValue(""); // Focus
        await element.addValue("FastText_");
        await sleep(1000);

        console.log("2. Setting global delay using extension (testing various formats)");
        await driver.execute('windows: typeDelay', { delay: 500 });
        await driver.execute('windows: typeDelay', { delay: '500' });
        await driver.execute('windows: typeDelay', 500);
        await driver.execute('windows: typeDelay', '500');

        console.log("   Typing... (should be slow)");
        await element.addValue("SlowText_");
        await sleep(1000);

        console.log("3. Using inline delay [delay:10] override - Should be fast");
        // Note: The inline delay only applies to this specific action
        await element.addValue("[delay:10]FastInline_");
        await sleep(1000);

        console.log("4. Verifying global delay (500ms) persists after inline action - Should be slow again");
        await element.addValue("SlowAgain_");

    } catch (error) {
        console.error(error);
    } finally {
        await driver.deleteSession();
    }
}

main();

File: examples_bak/basics/type_delay_test.py
from appium import webdriver
from appium.options.common.base import AppiumOptions
import time

def main():
    options = AppiumOptions()
    options.load_capabilities({
        "appium:automationName": "NovaWindows2",
        "platformName": "Windows",
        "appium:app": "Root",
        "appium:typeDelay": 500  # Start with slow global delay (500ms)
    })

    driver = webdriver.Remote(
        command_executor='http://172.16.1.52:4723',
        options=options
    )

    try:
        element = driver.find_element(by='xpath', value="//Document[@Name='Text Editor']")
        element.click() # Ensure focus
        
        print("1. Typing with configured delay (500ms) - Should be slow")
        element.send_keys("Slow_")
        time.sleep(1)

        print("2. Using inline delay [delay:10] to override global delay - Should be FAST")
        # This tests the user's specific request: global > 0 but inline is fast
        element.send_keys("[delay:10]Fast_")
        time.sleep(1)

        print("3. Verifying global delay (500ms) persists - Should be slow again")
        element.send_keys("SlowAgain_")
        time.sleep(1)

        print("4. Changing global delay to 0ms using extension")
        driver.execute_script('windows: typeDelay', {'delay': 0})
        
        print("   Typing... (should be fast)")
        element.send_keys("FastFinal_")

    except Exception as e:
        print(f"Error: {e}")
    finally:
        driver.quit()

if __name__ == "__main__":
    main()
